framework/src/utils/PetscDMMoose.C

```diff
@@ -631,7 +632,7 @@ DMMooseGetEmbedding_Private(DM dm, IS * embedding)
               PenetrationInfo * pinfo = lit->second;
               std::map<dof_id_type, std::vector<dof_id_type> > & node_to_elem_map =
               dmm->_nl->_fe_problem.mesh().nodeToElemMap();
-              if (pinfo && pinfo->isCaptured())
+              if (pinfo && (pinfo->isCaptured() || dmm->_include_all_contact_nodes))
```

Comment:
DMMoose should really be agnostic to anything about contact.  You're using a "leaky abstraction" here.  Anything contact related should be only in the `ContactSplit`.